apps/designer/app/routes/canvas/$projectId.$pageId.tsx

```diff
@@ -0,0 +1,2 @@
+export * from "./$projectId";
+export { default } from "./$projectId";
```

Comment:
I have done something similar in the past, but I think this is wrong, because we are not supposed to be importing these manually. I wonder if pageId can be just optional?

Reply:
Remix doesn't seem to support optional dynamic segments. This solution seems fine to me, but I can change it if there're other way to achieve this.

Reply:
Indeed, there is an issue and a discussion but no implementation yet, that's weird, one thing I found is "splats"
https://remix.run/docs/en/v1/guides/routing#splats

Dunno if that's usable in a good way or if its better, e.g. can you have $projectId.*.tsx ? idk

Reply:
We could use splats, but then something like `/canvas/123/123/not_a_parameter_we_expect` will also be directed into our route and we'll have to produce 404 manually. Why not keep as is?

Reply:
yeah, that would be not ideal either